design-documents/0006-metadata-api/text.md

```diff
@@ -0,0 +1,118 @@
+# Defining Communication between the Portal and the Explorer
+
+## Metadata API
+
+**Authors:** [Madison Dunitz](mailto:madison.dunitz@chanzuckerberg.com)
+
+**Approvers:** [Trent Smith](mailto:trent.smith@chanzuckerberg.com), [Brian Raymor](mailto:braymor@chanzuckerberg.com), [Arathi Mani](mailto:arathi.mani@chanzuckerberg.com) [Bruce Martin](mailto:bruce@chanzuckerberg.com)
+
+## tl;dr
+
+The Explorer needs a way to gather data about a dataset, including what collection the dataset belongs to, that is not or can not be stored in the dataset cxg file. This requires allowing the explorer to look up information on a dataset based on its identifier via an endpoint in the portal API.
+
+## Problem Statement | Background
+
+Currently the Explorer doesn't know anything about the Data Portal. This keeps the coupling between the two systems loose, which is helpful. But we do need some level of integration and communication between the two services. Currently this is only possible in one place: in the metadata of the serialized cxg in the Explorer bucket.
+
+The only information the Explorer has is the name of the s3 bucket passed in via the config and the dataset identifier passed in via the url. While it would theoretically be possible to grep for the Explorer url in the /collections API response in order to map back to the collection_id, this approach is fairly brittle and may not scale well.
+
+Currently datasets are uploaded to the portal as part of a private collection, they are then converted to a variety of formats (h5ad, loom, rds, cxg) and stored in an S3 bucket for download or use by the Explorer. Additionally metadata about the dataset and the collection it belongs to is stored in a relational database. The owner of the collection can publish it, making it accessible through the cellxgene site.
+
+When updating a public dataset (a dataset that belongs to a public collection, with publicized urls for viewing the datasets via the Explorer), it is currently necessary to overwrite the dataset in the S3 bucket in order to maintain the explorer urls, rather than simply redirecting to the new dataset location.
+
+This also complicates the process of deleting a dataset that has been published, while still communicating any relevant information to the end user. If the dataset is simply deleted from S3, the explorer will have no knowledge of it, and when someone attempts to navigate to the Explorer page for the deleted dataset, it can only display a 404, after looking for the cxg in the s3 bucket and finding nothing there.
+
+Additionally cxg's are more or less read-only. But the Portal needs to communicate new information about the dataset after it writes the cxg. For example, perhaps there should be an update to the title of the dataset. There is no way to do this (and communicate the change to the Explorer) without rewriting the entire cxg.
```

Comment:
I do not see how this design addresses this case. What is the intersection with the [CXG Data Format Specification](https://github.com/chanzuckerberg/cellxgene/blob/main/dev_docs/cxg.md#corpora-schema-encoding)?